synapse/handlers/room_member.py

```diff
@@ -79,7 +80,7 @@ def __init__(self, hs: "HomeServer"):
         self.account_data_handler = hs.get_account_data_handler()
         self.event_auth_handler = hs.get_event_auth_handler()
 
-        self.member_linearizer = Linearizer(name="member")
+        self.member_linearizer: Linearizer = Linearizer(name="member")
```

Comment:
This type hint shouldn't be necessary since it is declared as a `Linearizer`. 😕

Reply:
Not having that gave mypy errors about not being able to determine the type of `member_linearizer`. I wasn't able to track down what was making mypy unhappy so thought adding in the type hint was a good enough solution

Reply:
I *think* there's something weird going on here because it is an abstract class. Seems like a reasonable enough solution.